src/homebridge-roku.js

```diff
@@ -100,29 +102,58 @@ class RokuAccessory {
     return accessoryInfo;
   }
 
+  doesSupportWakeOnLan() {
+    return this.info.supportsWakeOnWlan === 'true';
+  }
+
   setupTelevision() {
     const television = new Service.Television(this.name);
 
     television
       .getCharacteristic(Characteristic.Active)
-      .on('get', (callback) => {
-        this.roku
-          .info()
-          .then((info) => {
-            const value =
-              info.powerMode === 'PowerOn'
-                ? Characteristic.Active.ACTIVE
-                : Characteristic.Active.INACTIVE;
-            callback(null, value);
-          })
-          .catch(callback);
+      .on('get', async (callback) => {
+        try {
+          const info = await pTimeout(this.roku.info(), 1000);
+
+          const value =
+            info.powerMode === 'PowerOn'
+              ? Characteristic.Active.ACTIVE
+              : Characteristic.Active.INACTIVE;
+          callback(null, value);
+        } catch (error) {
+          if (
+            error.constructor === pTimeout.TimeoutError &&
+            this.doesSupportWakeOnLan()
+          ) {
+            callback(null, Characteristic.Active.INACTIVE);
+            return;
+          }
+
+          callback(error);
+        }
       })
-      .on('set', (newValue, callback) => {
+      .on('set', async (newValue, callback) => {
         if (newValue === Characteristic.Active.ACTIVE) {
-          this.roku
-            .keypress('PowerOn')
-            .then(() => callback(null))
-            .catch(callback);
+          try {
+            await pTimeout(this.roku.keypress('PowerOn'), 1000);
+
+            callback(null);
+          } catch (error) {
+            if (
+              error.constructor === pTimeout.TimeoutError &&
+              this.doesSupportWakeOnLan()
+            ) {
+              if (this.info.ethernetMac && this.info.ethernetMac !== '') {
```

Comment:
Empty strings are falsy so the second check here is redundant, it can just be `if (this.info.ethernetMac)`

Looking at https://github.com/bschlenk/homebridge-roku/issues/9, it looks like there is both an `ethernetMac` and a `wifiMac`, do you know if those are usually the same value? Since we're checking for support for `wlan`, which I assume is wireless lan, we should probably be using the `wifiMac` here. Or am I missing something?

Reply:
b333242